src/app/caesar-portal/locations/locations.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit, TemplateRef, EventEmitter, Output } from '@angular/core';
+import { BsModalService } from 'ngx-bootstrap/modal';
+import { BsModalRef } from 'ngx-bootstrap/modal/modal-options.class';
+import { Response } from '@angular/http';
+import { Router } from '@angular/router';
+import { Location } from '../common/models/location';
+import { LocationService } from '../../caesar-portal/common/services/location.service';
+
+@Component({
+    selector: 'caesar-locations',
+    templateUrl: './locations.component.html',
+    styleUrls: ['./locations.component.scss']
+})
+export class LocationsComponent implements OnInit {
+    selectedLocatins: string[] = [];
+    locations: Location[];
+    modalRef: BsModalRef;
+    error: any;
+    
+    constructor(
+        private modalService: BsModalService,
+        private locationsService: LocationService,
+        private router : Router ) {}
+
+    @Output() onOpenModal = new EventEmitter();
+    private openModal(template: TemplateRef<any>): void {
+        this.modalRef = this.modalService.show(template, {class: 'delete-dialog'});
+    }
+
+    ngOnInit() {
+        this.locationsService.getLocations()
+              .subscribe(
+                  data => this.locations = data,
+                  (error) => {
+                      this.error = error;
+                      console.error(error);
+                  }
+              );
+    }
+
+    private closeModal(): void {
+        this.modalRef.hide();
+        this.selectedLocatins = [];
+    }
+
+    private confirmLocations(): void {
+        const result = this.selectedLocatins.join('+');
+        this.closeModal();
+        this.router.navigateByUrl(`${result}/groups`);
+    }
+
+    private addLocation($event, location): void {
```

Comment:
rename this method, it should be obvious by name what it does, right now it says "add locations" but it can remove it instead

Reply:
ok